core/src/commonMain/kotlin/com/xebia/functional/xef/prompt/evaluator/PromptEvaluator.kt

```diff
@@ -0,0 +1,161 @@
+    val scalability: Double,
+    val legalAndEthicalConsiderations: Double,
+    val reasoning: String,
+  )
+
+  suspend fun evaluate(
+    model: Chat,
+    scope: CoreAIScope,
+    prompt: String,
+    response: String,
+    promptConfiguration: PromptConfiguration = PromptConfiguration.DEFAULTS,
+  ): Score {
+    val result: List<String> =
+      model.promptMessages(
+        messages =
+          listOf(
+            Message.systemMessage {
+              // language=markdown
+              """
+          # PromptEvaluator
+          
+          # Roleplay as an evaluator for testing the performance of prompts over LLMs.
+          
+          EvaluationProcess {
+            State {
+              Prompt: String
+              Response: String
+            }
+            Completeness() {
+              Coverage()
+              Detail()
+            }
+            Accuracy() {
+              Correctness()
+              Consistency()
+            }
+            Efficiency() {
+              ResponseTime()
+              ResourceUtilization()
+            }
+            Robustness() {
+              NoiseTolerance()
+              Adaptability()
+            }
+            BiasAndFairness() {
+              DemographicBias()
+              ContentBias()
+            }
+            Interpretability() {
+              Clarity()
+              Explainability()
+            }
+            Usability() {
+              UserExperience()
+              Accessibility()
+            }
+            SafetyAndSecurity() {
+              ContentSafety()
+              DataSecurity()
+            }
+            CreativityAndFlexibility() {
+              Innovation()
+              Versatility()
+            }
+            Internationalization() {
+              LanguageSupport()
+              CulturalSensitivity()
+            }
+            Scalability() {
+              VolumeHandling()
+              Integration()
+            }
+            LegalAndEthicalConsiderations() {
+              Compliance()
+              EthicalConsiderations()
+            }
+            ScoreRules() {
+              Apply a score to each evaluation.
+              0 = not met
+              0.25 = partially met but not enough
+              0.5 = partially met
+              0.75 = mostly met
+              1 = met
+            }
+            Evaluate(prompt: String, response: String) : JSON {
+              Set Prompt = prompt
+              Set Response = response
+              ScoreRules() {
+                Completeness():evaluation=thorough
+                Accuracy():evaluation=precise
+                Efficiency():evaluation=optimized
+                Robustness():evaluation=resilient
+                BiasAndFairness():evaluation=unbiased
+                Interpretability():evaluation=clear
+                Usability():evaluation=user-friendly
+                Safety():evaluation=safe
+                Security():evaluation=secure
```

Comment:
Do you think we should merge these two? It's unclear to me how the LLM calculates this with one single abstract function definition (`SafetyAndSecurity`) and how is inferred in one value.

Reply:
It infers that `Safety` and `Security` are nested and hierarchical statements to accomplish `SafetyAndSecurity` but we can change this to whatever makes sense.